### PR TITLE
Allow 3rd party gambits, fix DM control and allow it to be hidden

### PR DESCRIPTION
**Pull request by CDK2020**
This allows 3rd party extensions to hook into the user search feature of Byobu to use their own gambit as well as hide the DM.

It also cleans up a bug caused by the allow-pd update.

### CODE REVIEW

File: js/src/forum/addDiscussPrivatelyControl.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default function () {
         if (app.session.user &&
             app.session.user.id() !== user.id() &&
             app.forum.attribute('canStartPrivateDiscussion') &&
-            (user.blocksPd() === false || app.forum.attribute('canStartPrivateDiscussionWithBlockers'))
+            ((user.blocksPd() === 0 || app.forum.attribute('canStartPrivateDiscussionWithBlockers')) && !user.cannotBeDirectMessaged())
```
Comment (luceos) [inline js/src/forum/addDiscussPrivatelyControl.js]: Fixed this https://github.com/FriendsOfFlarum/byobu/commit/ea4a64a47ec84b37c2576a500fbb656a19866a8a

Please revert the false => 0 change

**Comment by luceos** [comment]
What is the reasoning to change the name of the Gambit?

**Comment by CDK2020** [comment]
Since it's not just the allows-pd Gambit, (it now performs more functions) I think it makes sense to change the name. When looking at the request "byobu-search" tells you exactly what is going on.

**Comment by luceos** [comment]
I disagree. Because the Gambit is to interact with Byobu and the Gambit should filter users that Allow Private Discussions. Renaming it to suit the name of the package makes it unclear what the intent of the gambit is. So please revert, aside from that it looks good.

**Comment by CDK2020** [comment]
I see your point there. Reverted

**Comment by luceos** [comment]
Ps why did you prefix your branch with my initials 😂

**Pull request merged**